### ISSUE TITLE
A model for 2D pose estimation is not trained well on RTX 2080 Ti

### ISSUE THREAD

**Issue by grayish** [pytorch/pytorch#13663]
### Training results are not same on same codes between two GPUs, RTX 2080 Ti and GTX 1080 Ti.
I work on 2D pose estimation from a single image.
With RTX 2080 Ti, the stacked hourglass model shows the PCKh@0.5 score (accuracy measurement in pose estimation) at 4.58% after one day training. However, with GTX 1080 Ti, it shows the PCKh@0.5 score at 76.12% after 16 hours training.

The stacked hourglass model consists of nn.Conv2D, nn.BatchNorm2D, and nn.ReLU modules. In the training sequence, I used nn.MSELoss as criterion and optim.RMSprop as optimizer.

## To Reproduce

1. download MPII dataset (http://human-pose.mpi-inf.mpg.de/)
2. Clone bearpaw's pytorch-pose repository (https://github.com/bearpaw/pytorch-pose) with submodule
```
git clone --recursive https://github.com/bearpaw/pytorch-pose.git
```
3. Create a symbolic link to the images directory of the MPII dataset:
```
ln -s PATH_TO_MPII_IMAGES_DIR data/mpii/images
```
4. Run the following command in terminal to train the stacked hourglass model on MPII
```
CUDA_VISIBLE_DEVICES=0 python example/mpii.py -a hg --stacks 2 --blocks 1 --checkpoint checkpoint/mpii/hg_s2_b1
```

model code: https://github.com/bearpaw/pytorch-pose/blob/master/pose/models/hourglass.py
training code: https://github.com/bearpaw/pytorch-pose/blob/master/example/mpii.py

### [Computer1 - RTX2080Ti]
PyTorch version: 0.3.1.post2
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 18.04.1 LTS
GCC version: (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
CMake version: version 3.10.2

Python version: 2.7
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: 
GPU 0: GeForce RTX 2080 Ti
GPU 1: GeForce RTX 2080 Ti
(but I used only GPU 0.)

Nvidia driver version: 410.73
cuDNN version: Probably one of the following:
/usr/local/cuda-10.0/lib64/libcudnn.so.7.3.1
/usr/local/cuda-10.0/lib64/libcudnn_static.a
/usr/local/cuda-9.0/lib64/libcudnn.so.7.1.4
/usr/local/cuda-9.0/lib64/libcudnn_static.a

Versions of relevant libraries:
[pip] numpy (1.15.1)
[pip] numpydoc (0.8.0)
[pip] torch (0.3.1.post2)
[pip] torchvision (0.2.0)
[conda] cuda90                    1.0                  h6433d27_0    pytorch
[conda] pytorch                   0.3.1           py27_cuda9.0.176_cudnn7.0.5_2  [cuda90]  pytorch
[conda] torchvision               0.2.0            py27hfb27419_1    pytorch

### [Computer2 - GTX1080Ti]
PyTorch version: 0.3.1.post2
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 18.04.1 LTS
GCC version: (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
CMake version: Could not collect

Python version: 2.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 410.73
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy (1.15.1)
[pip] numpydoc (0.8.0)
[pip] torch (0.3.1.post2)
[pip] torchvision (0.2.0)
[conda] cuda90                    1.0                  h6433d27_0    pytorch
[conda] pytorch                   0.3.1           py27_cuda9.0.176_cudnn7.0.5_2  [cuda90]  pytorch
[conda] torchvision               0.2.0            py27hfb27419_1    pytorch

 - How you installed PyTorch (`conda`, `pip`, source): 
```
conda install -c pytorch pytorch=0.3.1 cuda90 torchvision
```

## Additional context

I did additional experiments on PyTorch 0.4.1 with CUDA 9.2 in conda environments. I migrated the model codes to PyTorch 0.4.1 and it was not trained at all.
Also, I built PyTorch 1.0  from the source with CUDA10. This version did not work well, too.

**Comment by zou3519** [comment]
You should open an issue in https://github.com/bearpaw/pytorch-pose. If you could investigate further and give us a smaller repro, that would be great.

**Comment by grayish** [comment]
Thanks for the advise. 
I tried to train the model without cudnn by "torch.backends.cudnn.enabled = False".
It was about two times slower, but I succeeded in training. So I also updated the Nvidia graphic driver to 415.13 and CUDNN version to 7.4.1. After rebuilding the PyTorch, this also succeeded in training.

I guess there are problems in CUDNN with Turing architecture because Nvidia said "While running cuDNN 7.3.0 on Turing with CUDA 10.0, r400 driver, the functions cudnnRNNForwardTraining(Ex) and cudnnRNNForwardInference(Ex) errored out returning CUDNN_STATUS_NOT_SUPPORTED." in their release note. (they said it was fixed. [https://docs.nvidia.com/deeplearning/sdk/cudnn-release-notes/rel_731.html#rel_731])
But I do not know why it succeeded in Tensorflow environment with CUIDA 9.0 and CUDNN 7.1.4.
